tests/README.md

```diff
@@ -0,0 +1,124 @@
+# tests
+
+This directory contains tests for various functionality areas of MicroPython.
+To run all stable tests, run "run-tests.py" script in this directory.
+
+Tests of capabilities not supported on all platforms should be written
+to check for the capability being present. If it is not, the test
+should merely output 'SKIP' followed by the line terminator, and call
+sys.exit() to raise SystemExit, instead of attempting to test the
+missing capability. The testing framework (run-tests.py in this
+directory, test_main.c in qemu_arm) recognizes this as a skipped test.
+
+There are a few features for which this mechanism cannot be used to
+condition a test. The run-tests.py script uses small scripts in the
+feature_check directory to check whether each such feature is present,
+and skips the relevant tests if not.
+
+Tests are generally verified by running the test both in MicroPython and
+in CPython and comparing the outputs. If the output differs the test fails
+and the outputs are saved in a .out and a .exp file respectively.
+For tests that cannot be run in CPython, for example because they use
+the machine module, a .exp file can be provided next to the test's .py
+file. A convenient way to generate that is to run the test, let it fail
+(because CPython cannot run it) and then copy the .out file (but not
+before checking it manually!)
+
+When creating new tests, anything that relies on float support should go in the
+float/ subdirectory.  Anything that relies on import x, where x is not a built-in
+module, should go in the import/ subdirectory.
+
+## perf_bench
+
+The `perf_bench` directory contains some performance benchmarks that can be used to benchmark different MicroPython firmwares or host ports.
+
+The runner utility is `run-perfbench,py`. Execute `./run-perfbench.py --help` for a full list of command line options.
+
+### Benchmarking a target
+
+To run tests on a firmware target using `pyboard.py`, run the command line like this:
+
+
+```
+./run-perfbench.py -p -d /dev/ttyACM0 168 100
+```
+
+* `-p` indicates running on a remote target via pyboard.py, not the host
+* `-d PORTNAME` is the serial port
+* `168` is value `N`, the approximate CPU frequency in MHz (in this case Pyboard V1.1 is 168MHz)
+* `100` is value `M`, the approximate heap size in kilobytes (can get this from `import micropython; micropython.mem_info()` or estimate it)
+
```

Comment:
Maybe mention that you're also free to pick other values, like 10/10, to get it to run faster, or 1000/100 to get it to run longer.  The main aim with value N is so that each test takes very roughly 1 second to complete (when run 8 times for averaging).  And tests are tuned so they can fit on a target with M-kbytes heap.